autotest/utilities/test_gdalalg_raster_proximity.py

```diff
@@ -0,0 +1,267 @@
+                "nodata": 0,
+            },
+            np.array(
+                [[0.0, 0.0, 2.0], [0.0, 1.4142135, 1.0], [2.0, 1.0, 0.0]],
+                dtype=np.float32,
+            ),
+        ),
+        # Test with target-values 1 and 3
+        (
+            {
+                "datatype": "Byte",
+                "target-values": [1, 3],
+                "distance-units": "PIXEL",
+                "max-distance": 2,
+                "nodata": 0,
+            },
+            np.array([[0, 1, 2], [1, 1, 1], [2, 1, 0]], dtype=np.uint8),
+        ),
+        # Test nodata 255
+        (
+            {
+                "target-values": [1],
+                "distance-units": "PIXEL",
+                "max-distance": 2,
+                "nodata": 255,
+            },
+            np.array([[255, 255, 2], [255, 1.4142135, 1], [2, 1, 0]], dtype=np.float32),
+        ),
+        # Test DISTANCE_UNITS=GEOGRAPHIC
+        (
+            {
+                "datatype": "Float32",
+                "target-values": [1],
+                "distance-units": "GEO",
+                "max-distance": 2,
+                "nodata": 0,
+            },
+            np.array([[0, 0, 2], [0, 1.4142135, 1], [2, 1, 0]], dtype=np.float32),
+        ),
+        # Test TILED creation option
+        (
+            {
+                "datatype": "Byte",
+                "target-values": [1],
+                "distance-units": "PIXEL",
+                "max-distance": 2,
+                "nodata": 0,
+                "creation-option": {"TILED": "YES"},
+            },
+            np.array([[0, 0, 2], [0, 1, 1], [2, 1, 0]], dtype=np.uint8),
+        ),
+        # Test fixed buffer value
+        (
+            {
+                "datatype": "Byte",
+                "target-values": [1],
+                "distance-units": "PIXEL",
+                "max-distance": 2,
+                "nodata": 255,
+                "fixed-buffer": 128,
+            },
+            np.array([[255, 255, 128], [255, 128, 128], [128, 128, 0]], dtype=np.uint8),
+        ),
+        # Test using band 2
+        (
+            {
+                "band": 2,
+                "datatype": "Byte",
+                "target-values": [1],
+                "distance-units": "PIXEL",
+                "max-distance": 2,
+                "nodata": 0,
+            },
+            np.array(
+                [
+                    [
+                        2,
+                        0,
+                        0,
+                    ],
+                    [1, 1, 0],
+                    [0, 1, 2],
+                ],
+                dtype=np.uint8,
+            ),
+        ),
+    ),
+)
+@pytest.mark.require_driver("GTiff")
+def test_gdalalg_raster_proximity_options(tmp_vsimem, options, expected_output_data):
+    """Test proximity calculation with several options."""
+    input_data = np.array([[3, 0, 0], [0, 0, 0], [0, 0, 1]], dtype=np.uint8)
+    src_filename = f"{tmp_vsimem}/prox_in.tif"
```

Comment:
These conversions to string should not be necessary (although they are harmless)

Reply:
can you please clarify?

Reply:
you can just do: ``src_filename = tmp_vsimem / "prox_in.tif"``